x-pack/plugins/task_manager/task_store.ts

```diff
@@ -84,47 +85,60 @@ export class TaskStore {
     this.supportedTypes = opts.supportedTypes;
   }
 
+  public addSupportedTypes(types: string[]) {
+    if (!this.wasInitialized) {
+      this.supportedTypes = this.supportedTypes.concat(types);
+    } else {
+      throw Error('Cannot add task types after initializattion');
+    }
+  }
+
   /**
    * Initializes the store, ensuring the task manager index is created and up to date.
    */
   public async init() {
-    const properties = {
-      type: { type: 'keyword' },
-      task: {
-        properties: {
-          taskType: { type: 'keyword' },
-          runAt: { type: 'date' },
-          interval: { type: 'text' },
-          attempts: { type: 'integer' },
-          status: { type: 'keyword' },
-          params: { type: 'text' },
-          state: { type: 'text' },
-          user: { type: 'keyword' },
-          scope: { type: 'keyword' },
+    if (!this.wasInitialized) {
```

Comment:
would you mind switching this to an early return?

Reply:
Not at all